src/nvim/regexp_defs.h

```diff
@@ -32,7 +32,16 @@
 
 // In the NFA engine: how many states are allowed.
 #define NFA_MAX_STATES 100000
-#define NFA_TOO_EXPENSIVE -1
+
+/// Result of a regex search.
+///
+/// kRegexMatchNotFound and kRegexMatchesFound may also be used for their
+/// integer values.
```

Comment:
It was to minimize changes, yes.
One function returns TRUE/FALSE (not the equivalent enum values) and calculates the number of matches.  Another function returns kRegexMatchTooCostly (-1), kRegexMatchNotFound (0), or the number of matches (>0).